pkg/fetch/vendir.go

```diff
@@ -7,26 +7,33 @@ import (
 	"bytes"
 	"fmt"
 	"io"
+	"strings"
 
 	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/config/global"
 	vendirconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	kyaml "sigs.k8s.io/yaml"
 )
 
-const vendirEntireDirPath = "."
+const (
+	vendirEntireDirPath     = "."
+	registryDefaultHostname = "index.docker.io"
+)
 
 type Vendir struct {
 	nsName     string
 	coreClient kubernetes.Interface
 	config     vendirconf.Config
+	kcConfig   *global.Config
 }
 
-func NewVendir(nsName string, coreClient kubernetes.Interface) *Vendir {
+func NewVendir(nsName string, coreClient kubernetes.Interface, globalConfig *global.Config) *Vendir {
```

Comment:
seems a little weird for this package to depend on some global config. should this be an interface?